### PR TITLE
Refactor for role-based db access

### PR DESCRIPTION
**Pull request by richtier**
[this task](https://uktrade.atlassian.net/browse/ED-1738)

### CODE REVIEW

File: directory_api_external/base.py [hunk 1]
```diff
@@ -20,57 +20,71 @@ def __init__(self, base_url=None, api_key=None):
         self.base_url = base_url
         self.request_signer = RequestSigner(secret=api_key)
 
-    def put(self, url, data):
+    def put(self, url, data, sso_session_id=None):
```
Comment (kowalcj0) [inline directory_api_external/base.py]: @richtier if `sso_session_id` is an optional argument, then maybe it'd be nice to make it truly optional like it's described here: 
http://www.asmeurer.com/python3-presentation/slides.html#9 (slides 9 to 31)

File: directory_api_external/base.py [hunk 2]
```diff
@@ -20,57 +20,71 @@ def __init__(self, base_url=None, api_key=None):
...
             url=url,
             method="PUT",
             content_type="application/json",
-            data=json.dumps(data)
+            data=json.dumps(data),
+            sso_session_id=sso_session_id,
         )
 
-    def patch(self, url, data, files=None):
+    def patch(self, url, data, files=None, sso_session_id=None):
```
Comment (kowalcj0) [inline directory_api_external/base.py]: ditto

File: directory_api_external/base.py [hunk 3]
```diff
@@ -20,57 +20,71 @@ def __init__(self, base_url=None, api_key=None):
...
                 method="PATCH",
                 content_type="application/json",
                 data=json.dumps(data),
+                sso_session_id=sso_session_id,
             )
         return response
 
-    def get(self, url, params=None):
-        return self.request(url=url, method="GET", params=params)
+    def get(self, url, params=None, sso_session_id=None):
```
Comment (kowalcj0) [inline directory_api_external/base.py]: ditto

File: directory_api_external/base.py (hunk 1 above)
Comment (richtier) [inline directory_api_external/base.py]: the usefulness seems to be "Oops, I accidentally passed too many arguments to the function, and one of them was swallowed by a keyword argument"

practically, we have tests that detect that - the tests call the client's functions and confirm the expected request was made.

good to know about that feature though :+1:

**Pull request merged**